### ISSUE TITLE
Test failure in emacsql-nul-character on Emacs 25

### ISSUE THREAD

**Issue by st3fan** [magit/emacsql#102]
While working on a test workflow (#101) I noticed that `emacsql-nul-character` is consistently failing on Emacs 25.x. The backtrace is a bit hard to decipher:

```
2022-10-09T14:41:03.1034160Z    passed  13/24  emacsql-insert
2022-10-09T14:41:03.1034493Z    passed  14/24  emacsql-limit
2022-10-09T14:41:03.1132341Z Test emacsql-nul-character backtrace:
2022-10-09T14:41:03.1313139Z   apply(#[771 "\301\302\303\304$A@\206
2022-10-09T14:41:03.1317660Z �\305C\"\207" [emacsql-sqlit
2022-10-09T14:41:03.1318476Z   emacsql-handle([eieio-class-tag--emacsql-sqlite-connection #<process
2022-10-09T14:41:03.1320050Z   #[257 "r\301!q\210eb\210p\302 \303\304\"\203�\305\302 \302 #\20
2022-10-09T14:41:03.1320714Z   apply(#[257 "r\301!q\210eb\210p\302 \303\304\"\203�\305\302 \30
2022-10-09T14:41:03.1321934Z   emacsql-parse([eieio-class-tag--emacsql-sqlite-connection #<process 
2022-10-09T14:41:03.1322689Z   #[642 "\300\301$\302!\210\303\"\210\304!\210\305!\207" [appl
2022-10-09T14:41:03.1323325Z   apply(#[642 "\300\301$\302!\210\303\"\210\304!\210\305!\207"
2022-10-09T14:41:03.1324027Z   emacsql([eieio-class-tag--emacsql-sqlite-connection #<process emacsq
2022-10-09T14:41:03.1324773Z   #[0 "\n\211\205Z�\211@\211A \303\304\305\306\307!\310\"\311$\216\
2022-10-09T14:41:03.1325485Z   ert--run-test-internal([cl-struct-ert--test-execution-info [cl-struc
2022-10-09T14:41:03.1326260Z   ert-run-test([cl-struct-ert-test emacsql-nul-character "Try insertin
2022-10-09T14:41:03.1327028Z   ert-run-or-rerun-test([cl-struct-ert--stats t [[cl-struct-ert-test e
2022-10-09T14:41:03.1327753Z   ert-run-tests(t #[385 "\306\307\"\203G�\211\211G\310U\203�\211@\20
2022-10-09T14:41:03.1328262Z   ert-run-tests-batch(nil)
2022-10-09T14:41:03.1328759Z   ert-run-tests-batch-and-exit()
2022-10-09T14:41:03.1329402Z   command-line-1(("-L" "../pg" "-L" "." "-L" "./tests" "-L" "tests" "-
2022-10-09T14:41:03.1330348Z   command-line()
2022-10-09T14:41:03.1330753Z   normal-top-level()
2022-10-09T14:41:03.1331261Z Test emacsql-nul-character condition:
2022-10-09T14:41:03.1331847Z     (emacsql-error "unrecognized token: \"'\"a\"")
2022-10-09T14:41:03.1333022Z    FAILED  15/24  emacsql-nul-character
2022-10-09T14:41:03.1334070Z    passed  16/24  emacsql-order-by
```

Is the `(emacsql-error "unrecognized token: \"'\"a\"")` a good hint?

**Comment by tarsius** [comment]
Please try running the tests locally. Note that it is a bit hard to tell for which an error occurs, so comment out all but `sqlite` in `emacsql-tests-connection-factories` and see if you can reproduce it like that.

I don't have an idea why that would not work for 25, while it does for other versions.

**Comment by tarsius** [comment]
I've just released v4.0.0 with new sqlite back-ends that require a newer emacs release. This older back-end is still included, but will be removed soon (because it is slower, less reliable and harder to maintain) at which point I'll have to depend on a newer emacs version.
